doc/user_guide/Parameters.ipynb

```diff
@@ -786,6 +786,191 @@
     "This approach can provide significant speedup and memory savings in certain cases, but should only be used for good reasons, since it can cause confusion for any code expecting instances to be independent as they have been declared."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "14a20588",
+   "metadata": {},
+   "source": [
+    "## Parameterized Abstract Base Class\n",
+    "\n",
+    "Param supports two ways of declaring a class as an abstract base class (ABC), which is a good approach to define an interface subclasses should implement:\n",
+    "\n",
+    "- Added in version 2.3.0, an abstract Parameterized class can be created by inheriting from `ParameterizedABC`, which is equivalent as inheriting from `ABC` from the Python [abc](https://docs.python.org/3/library/abc.html) module.\n",
+    "- A Parameterized class can be annotated with the class attribute `__abstract` set to `True` to declare it as abstract.\n",
+    "\n",
+    "We recommend adopting the first approach that is more generic and powerful. The second approach is specific to Param (`inspect.isabstract(class)` won't return `True` for example) and preserved for compatibility reasons.\n",
```

Comment:
Maybe it is worth adding a follow on sentence to strongly recommend *not* mixing approaches? Having two inconsistent ways of checking if a class is abstract sounds like a recipe for confusion!

That said, maybe param could have a wrapper utility that tries `inspect.isabstract(class)` first and if that returns `False`, checks again the param way?

Reply:
Comment added in https://github.com/holoviz/param/pull/1031/commits/ce27b13a529a98a40b5c5892cae15675e539d92a

> That said, maybe param could have a wrapper utility that tries inspect.isabstract(class) first and if that returns False, checks again the param way?

Internally, we have the `_is_abstract` function that does exactly that. I don't really want to make it public though:
- Parameterized classes have an `abstract` property (that's documented too, I checked) that accounts for both approaches
- The less public API the better
- Ultimately, the goal would be to deprecate and remove the old abstract way

If people feel strongly about it, we can add it in another release anyway.